lib/internal/policy/sri.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+// value of https://w3c.github.io/webappsec-subresource-integrity/#the-integrity-attribute
+
+// returns [{algorithm, value (in base64 string), options,}]
+const {
+  ERR_SRI_PARSE
+} = require('internal/errors').codes;
+const WSP = '[\\x20\\x09]';
+const VCHAR = '[\\x21-\\x7E]';
+const HASH_ALGO = 'sha256|sha384|sha512';
+// base64
+const HASH_VALUE = '[A-Za-z0-9+/]+[=]{0,2}';
+const HASH_EXPRESSION = `(${HASH_ALGO})-(${HASH_VALUE})`;
+const OPTION_EXPRESSION = `(${VCHAR}*)`;
+const HASH_WITH_OPTIONS = `${HASH_EXPRESSION}(?:[?](${OPTION_EXPRESSION}))?`;
+const SRI_PATTERN = new RegExp(`(${WSP}*)(?:${HASH_WITH_OPTIONS})`, 'g');
+Object.seal(SRI_PATTERN);
+const ALL_WSP = new RegExp(`^${WSP}*$`);
+Object.seal(ALL_WSP);
+const RegExp$exec = Function.call.bind(RegExp.prototype.exec);
+const RegExp$test = Function.call.bind(RegExp.prototype.test);
+const String$slice = Function.call.bind(String.prototype.slice);
+const {
+  Buffer: {
+    from: Buffer_from
+  }
+} = require('buffer');
+const {
+  Object: {
+    defineProperty
+  }
+} = global;
+const parse = (str) => {
+  SRI_PATTERN.lastIndex = 0;
+  let prevIndex = 0;
+  let match = RegExp$exec(SRI_PATTERN, str);
+  const entries = [];
+  while (match) {
+    if (match.index !== prevIndex) {
+      throw new ERR_SRI_PARSE(str, prevIndex);
+    }
+    if (entries.length > 0) {
+      if (match[1] === '') {
+        throw new ERR_SRI_PARSE(str, prevIndex);
+      }
+    }
+    // avoid setters being fired
+    defineProperty(entries, entries.length, {
+      enumerable: true,
+      configurable: true,
+      value: {
+        __proto__: null,
```

Comment:
There are quite a few `__proto__`. I tend to prefer `Object.create(null)` as it's more idiomatic.